cmd/charm/charmcmd/cmd.go

```diff
@@ -181,7 +181,9 @@ func addAuthFlag(f *gnuflag.FlagSet, s *string) {
 
 // addChannelFlag adds the -c (--channel) flags to the given flag set.
 func addChannelFlag(f *gnuflag.FlagSet, s *string) {
-	f.StringVar(s, "c", string(params.StableChannel), "the channel the charm or bundle is assigned to")
+	f.StringVar(s, "c",
+		fmt.Sprint(string(params.StableChannel), `|`, string(params.DevelopmentChannel)),
```

Comment:
+1 to mhilton's remark. Also, I have realised since this code was written that we really don't want to have "stable" as a default, because it's really not the same as specifying no channel. Specifically if you mention a charm with a revision number and don't specify a channel, the appropriate channel is chosen for you, which is useful behaviour.

So I'd suggest:

     f.StringVar(s, "c", "", fmt.Sprintf("the channel the charm or bundle is assigned to (%s|%s|%s)", params.StableChannel, params.DevelopmentChannel, params.UnpublishedChannel))